eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/list/Lists.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2017 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.api.factory.list;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+/**
+ * This class should be used to create instances of MutableList, ImmutableList and FixedSizeList
+ * <p>
+ * Mutable Examples:
+ *
+ * <pre>
+ * MutableList&lt;String&gt; emptyList = Lists.mutable.empty();
+ * MutableList&lt;String&gt; listWith = Lists.mutable.with("a", "b", "c");
+ * MutableList&lt;String&gt; listOf = Lists.mutable.of("a", "b", "c");
+ * </pre>
+ *
+ * Immutable Examples:
+ *
+ * <pre>
+ * ImmutableList&lt;String&gt; emptyList = Lists.immutable.empty();
+ * ImmutableList&lt;String&gt; listWith = Lists.immutable.with("a", "b", "c");
+ * ImmutableList&lt;String&gt; listOf = Lists.immutable.of("a", "b", "c");
+ * </pre>
+ *
+ * FixedSize Examples:
+ *
+ * <pre>
+ * FixedSizeList&lt;String&gt; emptyList = Lists.fixedSize.empty();
+ * FixedSizeList&lt;String&gt; listWith = Lists.fixedSize.with("a", "b", "c");
+ * FixedSizeList&lt;String&gt; listOf = Lists.fixedSize.of("a", "b", "c");
+ * </pre>
+ */
+@SuppressWarnings("ConstantNamingConvention")
+public final class Lists
+{
+    public static final ImmutableListFactory immutable;
+    public static final MutableListFactory mutable;
+    public static final FixedSizeListFactory fixedSize;
+
+    static
+    {
+        immutable = loadServiceClass(ImmutableListFactory.class);
+        mutable = loadServiceClass(MutableListFactory.class);
+        fixedSize = loadServiceClass(FixedSizeListFactory.class);
+    }
+
+    private Lists()
+    {
+        throw new AssertionError("Suppress default constructor for noninstantiability");
+    }
+
+    private static <T> T loadServiceClass(Class<T> serviceClass)
+    {
+        List<T> factories = new ArrayList<>();
+        for (T factory : ServiceLoader.load(serviceClass))
```

Comment:
Here's the ServiceLoader stuff. On failure, I could probably give better error messages.

Reply:
See my comment below. Maybe on failure we can have a default factory that throws when called (and a warning message at load time)?